musl doesn't seem to recognize UTC0 as a timezone. UTC works though.

TODO: Figure out if musl /should/ regonize UTC0.

diff -upr busybox-1.35.0.orig/testsuite/date/date-timezone busybox-1.35.0/testsuite/date/date-timezone
--- busybox-1.35.0.orig/testsuite/date/date-timezone	2022-03-02 22:53:12.337143305 +0100
+++ busybox-1.35.0/testsuite/date/date-timezone	2022-03-02 22:53:24.047137763 +0100
@@ -1,17 +1,17 @@
 # FEATURE: CONFIG_FEATURE_TIMEZONE
 
 # 'Z' is UTC
-dt=$(TZ=UTC0 busybox date -d '1999-1-2 3:4:5Z')
+dt=$(TZ=UTC busybox date -d '1999-1-2 3:4:5Z')
 dt=$(echo "$dt" | cut -b1-19)
 test x"$dt" = x"Sat Jan  2 03:04:05"
 
 # '+0600' is six hours ahead of UTC
-dt=$(TZ=UTC0 busybox date -d '1999-1-2 3:4:5 +0600')
+dt=$(TZ=UTC busybox date -d '1999-1-2 3:4:5 +0600')
 dt=$(echo "$dt" | cut -b1-19)
 test x"$dt" = x"Fri Jan  1 21:04:05"
 
 # '-0600' is six hours behind UTC
-dt=$(TZ=UTC0 busybox date -d '1999-1-2 3:4:5 -0600')
+dt=$(TZ=UTC busybox date -d '1999-1-2 3:4:5 -0600')
 dt=$(echo "$dt" | cut -b1-19)
 test x"$dt" = x"Sat Jan  2 09:04:05"
 
